src/libraries/Microsoft.PowerFx.Core/Public/Config/SymbolTable.cs

```diff
@@ -67,27 +40,52 @@ private void ValidateName(string name)
             {
                 throw new ArgumentException("Invalid name: ${name}");
             }
-        }
-
+        }
+
+        private int _slotCounter = 0;
+
         /// <summary>
         /// Provide variable for binding only.
         /// Value must be provided at runtime.
         /// </summary>
         /// <param name="name"></param>
-        /// <param name="type"></param>
-        public void AddVariable(string name, FormulaType type)
-        {
+        /// <param name="type"></param>
+        /// <param name="mutable"></param>
+        /// <param name="displayName"></param>
+        public ISymbolSlot AddVariable(string name, FormulaType type, bool mutable = false, string displayName = null)
```

Comment:
>name [](http://example.com/codeflow?start=46&length=4)

should we throw if the name is invalid? (null, only spaces, starts with a digit, only weird characters ...)

Reply:
ValidateName will on line 65